TMDBApp/TMDBApp/ListMovieViewController.swift

```diff
@@ -0,0 +1,177 @@
+            
+            self?.getMovies()
+        }
+    }
+    
+    fileprivate func getMovies(){
+        
+        self.showProgress()
+        
+        Requests.sharedInstance().requestMovies(fromPage: currentPage, withQuery: query) { [weak self] (result, error) in
+            
+            guard result != nil, let resultDict = result as? DictionaryData else{
+                self?.showError(error.debugDescription)
+                return
+            }
+            self?.parseMovie(resultDict)
+            
+            self?.tableView?.reloadData()
+            
+            self?.dismissProgress()
+        }
+    }
+    
+    fileprivate func parseGenrer(_ data:DictionaryData){
+        guard let results = data[jsonResult.genres.rawValue] as? [DictionaryData] else {
+            return
+        }
+        for result in results{
+            let genrer = Genrer(withDict: result)
+            genres.append(genrer)
+        }
+    }
+    
+    fileprivate func parseMovie(_ data:DictionaryData){
+        guard let results = data[jsonResult.results.rawValue]    as? [DictionaryData],
+              let total   = data[jsonResult.totalPages.rawValue] as? Int else {
+            return
+        }
+        totalPages = total
+        
+        for result in results{
+            let movie = Movie(withDict: result, andGenres: self.genres)
+            movies.append(movie)
+        }
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    func showError(_ error: String){
+        print(error)
+        updatesOnMain{
+            self.dismissProgress()
+            
+            let alert = UIAlertController(title: "Error!", message: "Check your internet connection. An error has occurred on getting data \(error)", preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "OK", style: .destructive, handler: nil))
+            self.present(alert, animated: true, completion: nil)
+
+        }
+    }
+    
+    // MARK: Navigation
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == Segues.detailMovieSegue.rawValue  {
+            if let viewController = segue.destination as? DetailMovieViewController {
+                if let movie = sender as? Movie{
+                    viewController.movie = movie
+                }
+            }
+        }
+    }
+}
+
+extension ListMovieViewController: UITableViewDelegate, UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return (movies.count == 0) ? 1 : movies.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        if movies.count > 0 {
```

Comment:
Is there any other possible approach instead of nesting the whole method functionality inside an if statement? In other words, how do you remove this `if` and still maintain the same logic?